business/workloads.go

```diff
@@ -2207,6 +2238,10 @@ func (in *WorkloadService) GetWorkloadAppName(ctx context.Context, cluster, name
 		return "", err
 	}
 
+	if wkd.IsGateway() || wkd.IsWaypoint() {
```

Comment:
I'm not 100% sure if we should do this. Waypoints act as gateways an their implementation uses actually a k8s gateway, but, they are not the same. For example, the proxy status is not checked for gateways but it should be checked for waypoints, or in this case, getAllGateways, https://github.com/kiali/kiali/blob/master/business/workloads.go#L181 should return also waypoints? 
As they don't have the same role in the infraestructure, maybe there are other implications about changing this. 
Maybe we can create a separate issue to revisit all this implications, maybe in other places we should change to logic to be if isGateway but !isWaypoint.

Reply:
OK, if we don't want to apply general gateway logic to waypoints then this would be a bad idea.  So, feel free to keep it as is, although we may want a comment on `IsGateway()` stating that it does not include waypoints, and to use `IsWaypoint()` for that use case.